workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/BodyAndModalsContainer.kt

```diff
@@ -26,32 +29,80 @@ internal class BodyAndModalsContainer @JvmOverloads constructor(
   defStyle: Int = 0,
   defStyleRes: Int = 0
 ) : FrameLayout(context, attributeSet, defStyle, defStyleRes) {
-
   private val baseViewStub: WorkflowViewStub = WorkflowViewStub(context).also {
     addView(it, ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT))
   }
 
-  private val dialogs = LayeredDialogs(this)
+  private val dialogs = LayeredDialogs(view = this, modal = true)
+
+  private val bounds = MutableStateFlow(Bounds())
+  private val boundsRect = Rect()
+  private val boundsListener = OnGlobalLayoutListener {
+    getGlobalVisibleRect(boundsRect)
+    if (!boundsRect.matches(bounds.value)) bounds.value = boundsRect.toBounds()
+  }
```

Comment:
here we are just extracting the 'Global Visible Rect' right? - what does that represent?

Reply:
It's the bounds of the view in global coordinates -- the screen's coordinate system, not the parent view's. But I just noticed that this method returns false "If the view is completely clipped or translated out". I imagine the contents of the `Rect` are nonsense in that case.

Not even sure what to do if that happens. Probably should close all the dialogs.

Reply:
Ok. Sounds good.

Reply:
```kotlin
  // The bounds of this view in global (display) coordinates, as reported
  // by getGlobalVisibleRect.
  //
  // Made available to managed ModalScreenOverlayDialogFactory instances
  // via the ModalArea key in ViewEnvironment. When this updates their
  // updateBounds methods will fire. They should resize themselves to
  // avoid covering peers of this view.
  private val bounds = MutableStateFlow(Rect())
  private val boundsRect = Rect()

  private val boundsListener = OnGlobalLayoutListener {
    if (getGlobalVisibleRect(boundsRect) && boundsRect != bounds.value) {
      bounds.value = Rect(boundsRect)
    }
    // Should we close the dialogs if getGlobalVisibleRect returns false?
    // https://github.com/square/workflow-kotlin/issues/599
  }
```

Reply:
Note that I punted handling a `false` return in any interesting why. We've been shipping the equivalent code for more than a year and it's not been an issue. I'd like to dig into that case separately and carefully -- see if it's actually an issue at all.